samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -72,7 +96,10 @@ public ContainerManager(SamzaApplicationState samzaApplicationState, ClusterReso
    */
   void handleContainerLaunch(SamzaResourceRequest request, String preferredHost, SamzaResource allocatedResource,
       ResourceRequestState resourceRequestState, ContainerAllocator allocator) {
-    if (this.standbyContainerManager.isPresent()) {
+    if (getControlActionMetadata(request.getProcessorId()).isPresent()) {
+      handleContainerAllocationForExistingControlAction(request.getProcessorId(), allocator, resourceRequestState,
```

Comment:
Is this simply checkControlActionAndRunStreamProcessor ?
similar to checkStandbyConstraintsAndRunStreamProcessor.